src/Core/src/Platform/Windows/ButtonExtensions.cs

```diff
@@ -20,5 +29,93 @@ public static class ButtonExtensions
 
 		public static void UpdateFont(this MauiButton nativeButton, IButton button, IFontManager fontManager) =>
 			nativeButton.UpdateFont(button.Font, fontManager);
+
+		public static async Task UpdateImageSourceAsync(this MauiButton nativeButton, IButton button, IImageSourceServiceProvider? provider)
+		{
+			var imageSource = button.ImageSource;
+			var text = button.Text;
+
+			ImageSource? nativeImageSource = null;
+
+			// Get native image
+			if (imageSource != null && provider!= null)
+			{
+				var service = provider.GetRequiredImageSourceService(imageSource);
+				var result = await service.GetImageSourceAsync(imageSource);
+				nativeImageSource = result?.Value;
+			}
+
+			// No image, just the text
+			if (nativeImageSource == null)
+			{
+				nativeButton.Content = text;
+				button?.InvalidateMeasure();
+				return;
+			}
+
+			var imageSourceSize = nativeImageSource.GetImageSourceSize();
+
+			var image = new WImage
+			{
+				Source = nativeImageSource,
+				VerticalAlignment = VerticalAlignment.Center,
+				HorizontalAlignment = HorizontalAlignment.Center,
+				Stretch = WStretch.Uniform,
+				Width = imageSourceSize.Width,
+				Height = imageSourceSize.Height
+			};
+
+			// BitmapImage is a special case that has an event when the image is loaded
+			// when this happens, we want to resize the button
+			if (nativeImageSource is BitmapImage bitmapImage)
+			{
+				bitmapImage.ImageOpened += (sender, args) =>
+				{
+					var actualImageSourceSize = nativeImageSource.GetImageSourceSize();
```

Comment:
This is the code we have actually in Xamarin.Forms, however, I think we should use the height value of the Button and adapt the image to the size of the button, instead of using the size of the image and adapt the size of the button.